### DISCUSSION TITLE
Can't load pastefromhtml plugin due to No module named 'html.parser'

### DISCUSSION THREAD

**Discussion by mquan86** [zim-desktop-wiki/zim-desktop-wiki#2145] [Q&A]
Hi,

I have Zim 0.74.3 installed as portable (Windows 10) and I'm trying install plugin pastefromhtml (https://github.com/cetinkaya/pastefromhtml).
However in Plugin tab, say fail to load the plugin ("There was a problem loading this plugin")

In debug log it seems fail or missing module 'html.parser'?
Do you have any idea how to fix it?

```
ERROR: Could not load plugin pastefromhtml
Traceback (most recent call last):
  File "build/venv/lib/python3.9/site-packages/zim/gui/preferencesdialog.py", line 403, in __init__
  File "build/venv/lib/python3.9/site-packages/zim/plugins/__init__.py", line 473, in get_plugin_class
  File "build/venv/lib/python3.9/site-packages/zim/utils.py", line 36, in get_module
  File "C:\Portable\data\zim\plugins\pastefromhtml\__init__.py", line 20, in <module>
    from .htmlcdparser import HTMLCDParser
  File "C:\Portable\data\zim\plugins\pastefromhtml\htmlcdparser.py", line 15, in <module>
    from html.parser import HTMLParser
ModuleNotFoundError: No module named 'html.parser'
```

**Comment by jaap-karssenberg** [comment]
Afraid this is a third party plugin, so not supported by this bug tracker.
Given the error message I think there are dependencies not met in the
installation

Regards,
Jaap

Op ma 29 aug. 2022 03:14 schreef Minh Quan Pham ***@***.***>:

**Reply by mquan86** [reply]
Ok, I will then open a bug ticket in the plugin repo.

From the error I can see that it missing html.parser module, but my installed python does have it.
So I'm not sure Zim really use my installed python or it has embed one somehow, that is why I open the discussion here as think it is belong to the main application problem.

**Reply by mquan86** [reply]
https://github.com/cetinkaya/pastefromhtml/issues/12

**Reply by lorenrosen** [reply]
It's probably an issue with the python search path, and/or how the module is installed. I've had various problems with this. A short-term fix would be to explicitly add it to sys.path or PYTHONPATH
See, e.g. python - adding directory to sys.path /PYTHONPATH - Stack Overflow
 https://stackoverflow.com/questions/16114391/adding-directory-to-sys-path-pythonpath
though this may be outdated

**Reply by mquan86** [reply]
I set PYTHONPATH or sys.path but it doesn't work.
